var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -7,17 +7,114 @@
 from spack import *
 
 
-class Hpx(CMakePackage):
+class Hpx(CMakePackage, CudaPackage):
     """C++ runtime system for parallel and distributed applications."""
 
     homepage = "http://stellar.cct.lsu.edu/tag/hpx/"
-    url      = "http://stellar.cct.lsu.edu/files/hpx_1.0.0.tar.gz"
+    url = "https://github.com/STEllAR-GROUP/hpx/archive/1.2.1.tar.gz"
 
-    version('1.0.0', '4983e7c6402417ec794d40343e36e417')
+    version('1.2.1', sha256='8cba9b48e919035176d3b7bbfc2c110df6f07803256626f1dad8d9dde16ab77a')
+    version('1.2.0', sha256='20942314bd90064d9775f63b0e58a8ea146af5260a4c84d0854f9f968077c170')
+    version('1.1.0', sha256='1f28bbe58d8f0da600d60c3a74a644d75ac777b20a018a5c1c6030a470e8a1c9')
 
+    version('1.0.0', '4983e7c6402417ec794d40343e36e417', url='http://stellar.cct.lsu.edu/files/hpx_1.0.0')
+
+    variant('cxxstd',
+            default='17',
+            values=('98', '11', '14', '17'),
+            description='Use the specified C++ standard when building.')
+
+    variant(
+        'malloc', default='tcmalloc',
+        description='Define which allocator will be linked in',
+        values=('system', 'tcmalloc', 'jemalloc', 'tbbmalloc')
+    )
+
+    variant('instrumentation', values=any_combination_of(
+        'apex', 'google_perftools', 'papi', 'valgrind'
+    ), description='Add support for various kind of instrumentation')
+
+    variant('networking', default=True,
+            description='Support for networking and multi=node runs')
+    variant('tools', default=False, description='Build HPX tools')
+
+    depends_on('boost')
```

Comment:
This line isn't necessary with the line below.

Reply:
Isn't `boost` treated differently from `boost@1.55:` during normalization? I thought adding that would help the concretizer, but thinking twice I'm not sure anymore...

Reply:
Nope. This is handled differently though:
```python
depends_on('boost')
depends_on('boost@1.55:', when='@1.2.3:')
```
Maybe that's what you were thinking of.